libraries/SRV_Channel/SRV_Channel.cpp

```diff
@@ -137,6 +137,18 @@ void SRV_Channel::set_output_pwm(uint16_t pwm, bool force)
     }
 }
 
+// set normalised output from -1 to 1, assuming 0 at mid point of servo_min/servo_max
+void SRV_Channel::set_output_norm(float value)
+{
+    // constrain and correct direction of input value
+    value = constrain_float(value, -1.0f, 1.0f) * (get_reversed() ? -1.0f : 1.0f);
+
+    // convert normalised value to pwm
+    const uint16_t mid = (servo_max + servo_min) / 2;
+    const uint16_t range = (servo_max - servo_min);
+    set_output_pwm(constrain_int16(mid + (value * range * 0.5f), servo_min, servo_max));
```

Comment:
This whole function could be this?

```suggestion
    if (type_angle()) {
        pwm_from_angle(value * high_out)
    } else {
        pwm_from_range(value * high_out)
   }
    
```

Reply:
@IamPete1,

Yes, thanks.  I think that would work too the only issue is that it would be inconsistent with the get_output_norm().  I'm reluctant to change get_output_norm() because that would affect the KDECAN ESCs and I'm keen that we get something in so that we can get the walking robots support merged before the end of the GSoC period.

Reply:
I think your better of having a new function and leave the old one for KDECAN, for example, if I have a rover with a throttle 1000 is backwards, 1250 is stop and 2000 is forwards. So do set output norm from scripting with 0 and find I have 1/3rd throttle, and I have no way of finding what value I should use to stop.

Reply:
@IamPete1, OK, thanks.  Maybe I'll implement it as you've suggested and then just add some comments to say that get_output_norm() and set_output_norm() are inconsistent.

Reply:
@IamPete1, ah, the problem with this potential solution is that the high_out hasn't been set for the script1 outputs so we would need to add a set-range binding and then ask devs to call this and set it to something.  What they set it to wouldn't be too important but it would have to be set.

Reply:
We could default to angle 4500, https://github.com/ArduPilot/ardupilot/blob/8484e8271f686e259559fba62a874275bc4307d0/libraries/SRV_Channel/SRV_Channel_aux.cpp#L129

But also give the bindings to change it

Reply:
@IamPete1, I have never seen this defaulting before!  I had no idea it existed.

I'm happy with this idea, let's see what @WickedShell says.

Reply:
I'm fine with setting the default range on those to something, Personally I'd set it to range not angle, but I'm not that invested in which it goes to. I would suggest we can jump straight to using a much larger value then `4500` though and just give the scripting outputs more resolution off the bat :) (I realize our actual output layer is gonna truncate these down badly, but hey, one thing at a time :D )

Reply:
OK, i've gone ahead and done this now and re-tested and it all seems OK. txs!